src/shogun/multiclass/tree/RandomCARTree.cpp

```diff
@@ -108,18 +108,20 @@ int32_t CRandomCARTree::compute_best_attribute(const SGMatrix<float64_t>& mat, c
 			feats[j]=mat(idx[i],j);
 
 		// O(N*logN)
-		SGVector<index_t> sorted_args=CMath::argsort(feats);
+		SGVector<index_t> sorted_args(feats.size());
+		sorted_args.range_fill();
+		CMath::qsort_index(feats.vector, sorted_args.vector, feats.size());
 
 		// number of non-missing vecs
 		int32_t n_nm_vecs=feats.vlen;
-		while (feats[sorted_args[n_nm_vecs-1]]==MISSING)
```

Comment:
its set to `CMath::MAX_REAL_NUMBER` , an optional routine in case missing attribute values